vivo-project/Vitro
# Exclude context path from application generated links

**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues/3846)**: (please link to issue)
[VIVO PR](https://github.com/vivo-project/VIVO/pull/4011)
# What does this pull request do?
New boolean configuration property context.path.exclude to exclude tomcat application context from generated links useful in case the application is deployed with reverse proxy.

# What's new?
Created context.path.exclude configuration property to exclude context path from generated links. 
Replaced calls for context path to utility methods that allow overriding default context path.

# How should this be tested?
Default behavior when configuration property is not set should be the same.
* To test application behind proxy one would need to configure webserver and tomcat in proxy mode like described in workaround of linked issue and set  context.path.exclude = true in runtime.properties
* General navigation between application pages
* Authorization
* Edit this individual page should work the same

#Additional notes
* Duplicate tests were removed.
* This change require documentation to be updated.

# Interested parties
@VIVO-project/vivo-committers

# Reviewers' expertise
Candidates for reviewing this PR should have some of the following expertises:
1. Java

# Reviewers' report template
## General comment
A reviewer should provide here comments and suggestions for requested changes if any.
## Testing
A reviewer should briefly describe here how it was tested
## Code reviewing
A reviewer should briefly describe here which part was code reviewed
